scopa.py

```
@@ -0,0 +1,217 @@
+no_take = -99
+
+
+class Scopa:
+    # card deck
+    deck = []
+    # all hands
+    hands = []
+    # all piles
+    piles = []
+    # current table
+    table = []
+    # how many scopas
+    scopa_count = []
+    # how many points
+    points = []
+    # what was the last hand played
+    last_hand_played = -1
+
+    def __init__(self):
+        self.generate_deck()
+        # self.deck = ['03 ', '06 ', '05 ', '12 ', '01 ', '12*', '05*', '03 ', '02 ', '03 ', '11*', '05 ', '04 ', '03*', '01 ', '04*', '06*', '04 ', '02 ', '07 ', '13 ', '13*', '01*', '01 ', '07 ', '07*', '11 ', '07 ', '02 ', '12 ', '13 ', '06 ', '12 ', '11 ', '06 ', '04 ', '13 ', '02*', '05 ', '11 ']
+        self.initiate_play()
+
+    # this method generates random deck
+    def generate_deck(self):
+        all_cards = []
+        for color in colors:
+            for i in range(1, 8):
+                card = "0" + str(i) + color
+                all_cards.append(card)
+            for j in range(11, 14):
+                card = str(j) + color
+                all_cards.append(card)
+
+        self.deck = []
+        while len(all_cards) > 0:
+            pos = random.randint(0, len(all_cards) - 1)
+            random_card = all_cards[pos]
+            all_cards.remove(random_card)
+            self.deck.append(random_card)
+
+        print (self.deck)
+
+    # create table, hands, piles and scopa count
+    def initiate_play(self):
+        for i in range(0, initial_table_size):
+            self.table.append(self.deck.pop())
+
+        for p in range(0, no_of_players):
+            hand = []
+            self.draw_hand(hand)
+            self.hands.append(hand)
+            self.scopa_count.append(0)
+            self.piles.append([])
+
+    # draw cards from deck to hand, return False if hand is empty
+    def draw_hand(self, hand):
+        if len(self.deck) == 0:
+            return False
+        for c in range(0, hand_size):
+            hand.append(self.deck.pop())
+        return True

```

Instead of returning a boolean value, this method should throw an exception if the hand is empty